### PR TITLE
Go next after beating level

### LINKED ISSUE #287: [REFACTOR] Game flow refactor
## 🚧 Refactor

The game's flow feels a bit too long for the user to enjoy it. This could be improved doing the following flow:

### Implementation

- [ ] After winning each game the redirect button should go to the next level's lineup. "Next stage" button should be rename to "Next level".
- [ ] "Tap anywhere to return" should be rename as "Tap to return". The Victory / defeat splash should have a solid color background so that this text isn't as confusing for the user as it is now.
- [ ] Once all campaign levels are won the button should redirect the user to the level selection within the next campaign

### PR DESCRIPTION
**Pull request by tkz00**
Closes #287 

## Motivation

Having the user go back to campaign map each time a level was beaten instead of allowing him/her to go straight to the next level was bad and was detrimental to the overall user experience.

## Summary of changes

- Some texts of the victory/defeat splash screen have been updated to better convey the information they are supposed to communicate.
- The victory/defeat splash screen's backgrounds were changed to a fully opaque color, to improve visibility and readability.
- The next button in the victory splash takes the player directly to the lineup screen for the next level in order, except when the last level beaten is the last one of the respective campaign, in that case the user is taken back to the campaigns map to select a new campaign.

## How to test

Go through levels of the campaigns, when beaten, the Next button should take you to the lineup screen for the next level (this can be checked by leaving the lineup screen, reentering the available menu and checking if the enemy units are the same).

When the last level of the campaign is beaten, the "next campaign" button should take you to the map of all the campaigns.

The Victory / defeat splash should have a solid color background and the bottom text should be "Tap to return".

The blue button on the victory splash should have the text "NEXT LEVEL", or "NEXT CAMPAIGN" if the level beaten was the last one of it's campaign.

### CODE REVIEW

**Review by AngieDutra — CHANGES_REQUESTED** [review]
Tested and works as expected! Just a few comments:
Unity has a cool feature of "Safe area". This yellow line represents the area that will always display as visible and interactive in any phone. Leaving info outside this area is a risky move. I recommend moving the "Tap to return" text under the button. Although, leave a bit of space in between, so it doesn't get overpowered by the button in visual terms.
Captura de pantalla 2024-03-25 a la(s) 1 47 44 a  m

Also, watch out with this button size. The following shouldn't happen. Maybe add with to the button in this case. It's actually a different type of button that takes the user back to another scene. Maybe use the secondary button to move from each leven and use this primary one when the user defeats the boss and it's given the option to move to the next campaign 
Captura de pantalla 2024-03-25 a la(s) 1 51 49 a  m

**Comment by tkz00** [comment]
> Tested and works as expected! Just a few comments: Unity has a cool feature of "Safe area". This yellow line represents the area that will always display as visible and interactive in any phone. Leaving info outside this area is a risky move. I recommend moving the "Tap to return" text under the button. Although, leave a bit of space in between, so it doesn't get overpowered by the button in visual terms. Captura de pantalla 2024-03-25 a la(s) 1 47 44 a  m
> 
> Also, watch out with this button size. The following shouldn't happen. Maybe add with to the button in this case. It's actually a different type of button that takes the user back to another scene. Maybe use the secondary button to move from each leven and use this primary one when the user defeats the boss and it's given the option to move to the next campaign Captura de pantalla 2024-03-25 a la(s) 1 51 49 a  m

The "Tap to return" text has been moved inside the safe area, and the next campaign button text is handled in [this buttons redesign PR](https://github.com/lambdaclass/champions_of_mirra/pull/286).

**Review by AngieDutra — CHANGES_REQUESTED** [review]
Move the "Tap to return" up. It's still to low in the screen to be noticeable in time

**Review by AngieDutra — DISMISSED** [review]
Looks awesome! ✨

**Comment by martinacantaro** [comment]
Tested and all three changes are approved. 
The only thing is that the unit list in the battle screen overlaps with the Back and Battle buttons

Screenshot 2024-04-04 at 15 03 41

**Comment by tkz00** [comment]
> Tested and all three changes are approved. The only thing is that the unit list in the battle screen overlaps with the Back and Battle buttons
> Screenshot 2024-04-04 at 15 03 41

Good catch, fixed!

**Review by ncontinanza — CHANGES_REQUESTED** [review]

File: client/Assets/Scripts/Campaign/CampaignLevelsManager.cs [hunk 1]
```diff
@@ -16,9 +16,20 @@ public class CampaignLevelsManager : MonoBehaviour
...
     public void AssignLevelsData(List<LevelData> levelsData)
     {
-        // Get the min between the levels from the backend and the level indicators from the scene to assign to not go OutOfRange
-        for(int levelIndex = 0; levelIndex < Math.Min(levelsData.Count, levelIndicators.Count); levelIndex++) {
+		// The campaign prefabs need to match the number of levels with the levels brought with the backend, if not this will break.
+		if(levelsData.Count != levelIndicators.Count) {
+			throw new Exception("The number of levels brought from the backend doesn't match with the number of levels on that campaign prefab.");
+		}
+
+        for(int levelIndex = 0; levelIndex < levelsData.Count; levelIndex++) {
```
[the remark this answers is not in the document]
Comment (tkz00) [reply client/Assets/Scripts/Campaign/CampaignLevelsManager.cs:24]: Modified this

**Review by ncontinanza — APPROVED** [review]
LGTM!

**Pull request merged**